dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/datasource/redshift/RedshiftDataSourceProcessor.java

```diff
@@ -0,0 +1,140 @@
+        redshiftDatasourceParamDTO.setOther(parseOther(connectionParams.getOther()));
+
+        return redshiftDatasourceParamDTO;
+    }
+
+    @Override
+    public BaseConnectionParam createConnectionParams(BaseDataSourceParamDTO datasourceParam) {
+        RedshiftDataSourceParamDTO redshiftParam = (RedshiftDataSourceParamDTO) datasourceParam;
+        String address = String.format("%s%s:%s", Constants.JDBC_REDSHIFT, redshiftParam.getHost(), redshiftParam.getPort());
+        String jdbcUrl = address + "/" + redshiftParam.getDatabase();
+
+        RedshiftConnectionParam
+            redshiftConnectionParam = new RedshiftConnectionParam();
+        redshiftConnectionParam.setUser(redshiftParam.getUserName());
+        redshiftConnectionParam.setPassword(PasswordUtils.encodePassword(redshiftParam.getPassword()));
+        redshiftConnectionParam.setOther(transformOther(redshiftParam.getOther()));
+        redshiftConnectionParam.setAddress(address);
+        redshiftConnectionParam.setJdbcUrl(jdbcUrl);
+        redshiftConnectionParam.setDatabase(redshiftParam.getDatabase());
+        redshiftConnectionParam.setDriverClassName(getDatasourceDriver());
+        redshiftConnectionParam.setValidationQuery(getValidationQuery());
+        redshiftConnectionParam.setProps(redshiftParam.getOther());
+
+        return redshiftConnectionParam;
+    }
+
+    @Override
+    public ConnectionParam createConnectionParams(String connectionJson) {
+        return JSONUtils.parseObject(connectionJson, RedshiftConnectionParam.class);
+    }
+
+    @Override
+    public String getDatasourceDriver() {
+        return Constants.COM_REDSHIFT_JDBC_DRIVER;
+    }
+
+    @Override
+    public String getValidationQuery() {
+        return Constants.REDHIFT_VALIDATION_QUERY;
+    }
+
+    @Override
+    public String getJdbcUrl(ConnectionParam connectionParam) {
+        RedshiftConnectionParam
+            redshiftConnectionParam = (RedshiftConnectionParam) connectionParam;
+        if (!StringUtils.isEmpty(redshiftConnectionParam.getOther())) {
+            return String.format("%s?%s", redshiftConnectionParam.getJdbcUrl(), redshiftConnectionParam.getOther());
+        }
+        return redshiftConnectionParam.getJdbcUrl();
+    }
+
+    @Override
+    public Connection getConnection(ConnectionParam connectionParam) throws ClassNotFoundException, SQLException {
+        RedshiftConnectionParam redshiftConnectionParam = (RedshiftConnectionParam) connectionParam;
+        Class.forName(getDatasourceDriver());
+        return DriverManager.getConnection(getJdbcUrl(connectionParam),
+                redshiftConnectionParam.getUser(), PasswordUtils.decodePassword(redshiftConnectionParam.getPassword()));
+    }
+
+    @Override
+    public DbType getDbType() {
+        return DbType.REDSHIFT;
+    }
+
+    private String transformOther(Map<String, String> otherMap) {
+        if (MapUtils.isNotEmpty(otherMap)) {
+            List<String> list = new ArrayList<>();
```

Comment:
Suggest to add initialize size.
```suggestion
            List<String> list = new ArrayList<>(otherMap.size());
```